src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -221,7 +222,9 @@ else if (shouldCheckLastRcurly) {
         else if (shouldBeOnSameLine(bracePolicy, details)) {
             violation = MSG_KEY_LINE_SAME;
         }
-        else if (shouldBeAloneOnLine(bracePolicy, details)) {
+        else if (shouldBeAloneOnLine(bracePolicy, details)
+                || shouldBeAloneInLambda(bracePolicy, details)
+                || shouldBeAloneOrSLine(bracePolicy, details)) {
```

Comment:
shouldBeAloneOnLine is general question so all more specific question like shouldBeAloneInLambda and shouldBeAloneOrSLine should be part of general question (in side this method)